File: src/cpp-sample/Sample.cpp
#include <XtCpp.hpp>
#include <string>
#include <cstdlib>
#include <iostream>

extern int AggregateMain();
extern int FullDuplexMain();
extern int PrintSimpleMain();
extern int RenderSimpleMain();
extern int PrintDetailedMain();
extern int CaptureSimpleMain();
extern int RenderAdvancedMain();
extern int CaptureAdvancedMain();

static void RunSample(int32_t index)
{
  std::cout << Names[index] << ":\n";
  Samples[index]();
}

static const char* Names[] =
{
  "PrintSimple", "PrintDetailed", "CaptureSimple", "RenderSimple",
  "CaptureAdvanced", "RenderAdvanced", "FullDuplex", "Aggregate"
};

static int(*Samples[])() = 
{
  PrintSimpleMain, PrintDetailedMain, CaptureSimpleMain, RenderSimpleMain,
  CaptureAdvancedMain, RenderAdvancedMain, FullDuplexMain, AggregateMain, 
};

int main(int argc, char** argv) 
{
  int32_t index = argc == 2? std::stoi(std::string(argv[1])): -1;
  try
  {
    if (index >= 0) RunSample(index);
    else for (int32_t i = 0; i < sizeof(Samples)/sizeof(Samples[0]); i++) RunSample(i);
    return EXIT_SUCCESS;
  } catch (const Xt::Exception& e)
  { 
    std::cout << Xt::Audio::GetErrorInfo(e.GetError()) << "\n"; 
    return EXIT_FAILURE;
  }
}

File: src/java/sample/xt/sample/Sample.java
package xt.sample;

import xt.audio.XtAudio;
import xt.audio.XtException;

public final class Sample {

    interface Runner {
        void run() throws Exception;
    }

    static void runSample(int index) throws Exception {
        System.out.println(NAMES[index] + ":");
        SAMPLES[index].run();
    }

    static final String[] NAMES = new String[]{
            "PrintSimple", "PrintDetailed", "CaptureSimple", "RenderSimple",
            "CaptureAdvanced", "RenderAdvanced", "FullDuplex", "Aggregate"
    };

    static final Runner[] SAMPLES = new Runner[]{
            PrintSimple::main, PrintDetailed::main, CaptureSimple::main, RenderSimple::main,
            CaptureAdvanced::main, RenderAdvanced::main, FullDuplex::main, Aggregate::main
    };

    public static void main(String[] args) throws Exception {
        int index = args.length == 1? Integer.parseInt(args[0]): -1;
        try {
            if(index >= 0) runSample(index);
            else for(int i = 0; i < SAMPLES.length; i++) runSample(i);
        } catch(XtException e) {
            System.out.println(XtAudio.getErrorInfo(e.getError()));
        } catch(Throwable t) {
            System.out.println(t.getMessage());
        }
    }
}